### PR TITLE
SASS 2125 implemented workplace amount page

### PR DESCRIPTION
**Pull request by abiodxn**
### Description
Implement workplace amount page

[SASS 2125](https://jira.tools.tax.service.gov.uk/browse/SASS-2125)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [ ]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [x]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [x]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [x]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests? (where applicable)
- [ ]  Have you addressed warnings where appropriate?
- [ ]  Have you rebased against the current version of main?
- [ ]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

File: it/controllers/pensions/WorkplaceAmountControllerISpec.scala [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+          textOnPageCheck(hintText, hintTextSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          inputFieldValueCheck(amountInputName, inputSelector, existingAmount)
+          buttonCheck(buttonText, continueButtonSelector)
+          formPostLinkCheck(workplacePensionAmount(taxYearEOY), formSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+
+        //TODO redirect to question page
+        "redirect to the PensionsSummary page if the question has not been answered" which {
```
Comment (ILee15) [inline it/controllers/pensions/WorkplaceAmountControllerISpec.scala:218]: Can move the 3 redirect tests out of the foreach loop as its not testing any different content

File: it/controllers/pensions/WorkplaceAmountControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          inputFieldValueCheck(amountInputName, inputSelector, amountOverMaximum)
+          buttonCheck(buttonText, continueButtonSelector)
+          formPostLinkCheck(workplacePensionAmount(taxYearEOY), formSelector)
+          errorSummaryCheck(maxAmountErrorText, expectedErrorHref)
+          errorAboveElementCheck(maxAmountErrorText)
+          welshToggleCheck(user.isWelsh)
+        }
+
+        "redirect to the correct page when a valid amount is submitted and update the session amount" which {
```
Comment (ILee15) [inline it/controllers/pensions/WorkplaceAmountControllerISpec.scala:398]: Also can move this redirect test out of the foreach loop

File: app/models/redirects/ConditionalRedirect.scala [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ * limitations under the License.
+ */
+
+package models.redirects
+
+import play.api.mvc.Call
+
+case class ConditionalRedirect(condition: Boolean, redirect: Call, hasPrior: Option[Boolean] = None)
+
+object ConditionalRedirect {
```
Comment (ILee15) [inline app/models/redirects/ConditionalRedirect.scala:23]: Unless I've missed something this isn't used anywhere yet. Would it be better to include in a future task when needed?

**Review by ILee15 — CHANGES_REQUESTED** [review]

File: it/controllers/pensions/WorkplaceAmountControllerISpec.scala (hunk 2 above)
Comment (abiodxn) [reply it/controllers/pensions/WorkplaceAmountControllerISpec.scala:398]: Here we can test when the user is agent, and when the user is not agent, that's the reasoning for leaving in the for loop.

**Review by ILee15 — APPROVED** [review]

**Pull request merged**